src/overcooked_ai_py/agents/benchmarking.py

```diff
@@ -19,36 +20,76 @@ class AgentEvaluator(object):
         mlam_params (dict): the parameters for mlam, the MediumLevelActionManager
         debug (bool): whether to display debugging information on init
         """
-        assert type(mdp_params) is dict, "mdp_params must be a dictionary"
-        assert (mdp_params != {} and env_params != {}) or mdp_fn != None, "either evaluate from params or fn"
+        assert callable(mdp_fn), "mdp generating function must be a callable function"
         env_params["mlam_params"] = mlam_params
-        if mdp_fn is None:
+        self.mdp_fn = mdp_fn
+        self.env = OvercookedEnv(self.mdp_fn, **env_params)
+        self.force_compute = force_compute
+
+    @staticmethod
+    def from_mdp_params(mdp_params, env_params, outer_shape=None, mdp_params_schedule_fn=None, force_compute=False,
+                        mlam_params=NO_COUNTERS_PARAMS, debug=False):
+        """
+        mdp_params (dict): params for creation of an OvercookedGridworld instance through the `from_layout_name` method
+        outer_shape: the outer shape of environment
+        mdp_params_schedule_fn: the schedule for varying mdp params
+        Information for the rest of params please refer to the __init__ method above
+        """
+        # special case: generate from layout name
+        if type(mdp_params) is dict and "layout_name" in mdp_params:
             mdp = OvercookedGridworld.from_layout_name(**mdp_params)
-            self.mdp_fn = lambda: mdp
-            self.env = OvercookedEnv.from_mdp(mdp, **env_params)
-        else:
+            mdp_fn = lambda: mdp
+        else: # general case: variable mdp
+            assert outer_shape is not None, "outer_shape needs to be defined for variable mdp"
+            mdp_fn_naive = LayoutGenerator.mdp_gen_fn_from_dict(mdp_params, outer_shape, mdp_params_schedule_fn)
             # infinite mdp
-            if 'num_mdp' not in env_params or env_params['num_mdp'] == -1:
-                self.mdp_fn = mdp_fn
-                self.env = OvercookedEnv(self.mdp_fn, **env_params)
-            else:
+            if 'num_mdp' not in env_params or np.isinf(env_params['num_mdp']):
+                mdp_fn = mdp_fn_naive
+            else: # finite mdp, random choice
                 num_mdp = env_params['num_mdp']
-                assert num_mdp > 0, "invalid number of mdp"
-                self.mdp_lst = [mdp_fn() for _ in range(num_mdp)]
-                self.mdp_fn = lambda : np.random.choice(self.mdp_lst)
-                self.env = OvercookedEnv(self.mdp_fn, **env_params)
+                assert type(num_mdp) == int and num_mdp > 0, "invalid number of mdp: " + str(num_mdp)
+                mdp_lst = [mdp_fn_naive() for _ in range(num_mdp)]
+                mdp_fn = lambda : np.random.choice(mdp_lst)
+        if "outer_shape" in env_params:
+            del env_params["outer_shape"]
+        return AgentEvaluator(env_params, mdp_fn, force_compute, mlam_params, debug)
```

Comment:
I'm really sorry for not being clearer about this, but in general I feel like it's good practice to avoid a lot of conditional logic within methods (i.e. having one method do 3 different tasks), and instead have one method for each if/else condition (i.e. for each "special case"). In the `__init__` method this is particularly important, but I feel like this also applies to all other methods too. 

In this case, you can have one method to generate from a layout name, one for infinite variable mdp, and one for finite variable mdp (using a layout generator). Usually, there is a nice way to reuse some of the methods for others, creating some heirarchy of abstraction that enables you to not have to re-write the same lines 3 times.

In particular, looking at the other methods below, it seems like a method for single mdps already exists (`from_mdp`). If you want another method to create a AgentEvaluator from layout name, you could create a `from_layout_name` classmethod that calls `from_mdp` once you have created the MDP as `OvercookedGridworld.from_layout_name` (to avoid having code duplicates). And for the case of multiple MDPs created through layout_generator, you might want to create a method that is named more specifically invoking that purpose. You could probably also collapse the finite case into the infinite case, and remove another conditional statement, or if that turns out to be too hard, you can make them into two different classmethods.

Reply:
I see, thanks for the clarification!! This will be addressed in the newest commit.